### PR TITLE
test support for blackhole in Artery, #20589

### PR DESCRIPTION
**Pull request by patriknw**
We can do more fancy testing support later, but blackhole should make it possible to port most of the old tests.

Refs #20589

### CODE REVIEW

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -106,6 +106,9 @@ akka {
       # but must be resolved to ActorRefs first.
       large-message-destinations = []
       
+      # For enabling testing features, such as blackhole in akka-remote-testkit.
+      test-mode = off
```
Comment (drewhk) [inline akka-remote/src/main/resources/reference.conf]: maybe make it an advanced setting?

File: akka-remote/src/main/scala/akka/remote/artery/Association.scala [hunk 2]
```diff
@@ -85,6 +86,13 @@ private[remote] class Association(
   @volatile private[this] var _outboundControlIngress: OutboundControlIngress = _
   @volatile private[this] var materializing = new CountDownLatch(1)
 
+  private val _testStages: CopyOnWriteArrayList[TestManagementApi] = new CopyOnWriteArrayList
```
Comment (drewhk) [inline akka-remote/src/main/scala/akka/remote/artery/Association.scala]: Yeah, COW is perfect for such use-case

**Comment by drewhk** [comment]
LGTM

**Comment by johanandren** [comment]
LGTM

**Comment by patriknw** [comment]
rebased, will merge after validation

**Pull request merged**